File: typescript/tests/integration/beta-skills-session.integration.test.ts
// ci-stable
/**
 * Integration tests for session creation with skills loading.
 *
 * Note: This test calls the real backend. Do not run concurrently.
 */

import { AgentBay } from "../../src/agent-bay";
import { CreateSessionParams } from "../../src/session-params";
import { log } from "../../src/utils/logger";

describe("Beta skills session integration tests", () => {
  const apiKey = process.env.AGENTBAY_API_KEY;
  let agentBay: AgentBay;

  beforeEach(() => {
    if (apiKey) {
      agentBay = new AgentBay({ apiKey });
    }
  });

  test("should create session with loadSkills succeeds", async () => {
    if (!apiKey) {
      log("Skipping test: AGENTBAY_API_KEY not set");
      return;
    }

    const params = new CreateSessionParams({
      loadSkills: true,
    });
    const result = await agentBay.create(params);
    expect(result.success).toBe(true);
    expect(result.session).toBeDefined();

    const session = result.session!;
    try {
      expect(session.sessionId).toBeDefined();
    } finally {
      await agentBay.delete(session);
    }
  }, 120000);

  test("should create session without skills", async () => {
    if (!apiKey) {
      log("Skipping test: AGENTBAY_API_KEY not set");
      return;
    }

    const params = new CreateSessionParams({});
    const result = await agentBay.create(params);
    expect(result.success).toBe(true);
    expect(result.session).toBeDefined();

    const session = result.session!;
    try {
      expect(session.sessionId).toBeDefined();
    } finally {
      await agentBay.delete(session);
    }
  }, 120000);
});

File: golang/tests/pkg/integration/beta_skills_session_integration_test.go
// ci-stable
// Integration tests for session creation with skills loading.
//
// Note: This test calls the real backend. Do not run concurrently.
package integration_test

import (
	"testing"

	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay"
	"github.com/aliyun/wuying-agentbay-sdk/golang/tests/pkg/agentbay/testutil"
)

func TestBetaSkillsSession_CreateWithLoadSkillsSucceeds(t *testing.T) {
	apiKey := testutil.GetTestAPIKey(t)
	agentBay, err := agentbay.NewAgentBay(apiKey)
	if err != nil {
		t.Fatalf("Error initializing AgentBay client: %v", err)
	}

	params := agentbay.NewCreateSessionParams().WithLoadSkills(true)
	result, err := agentBay.Create(params)
	if err != nil {
		t.Fatalf("Create failed: %v", err)
	}

	if !result.Success {
		t.Fatalf("Create returned success=false")
	}
	if result.Session == nil {
		t.Fatal("Expected non-nil session")
	}

	session := result.Session
	defer func() {
		_, _ = agentBay.Delete(session)
	}()
}

func TestBetaSkillsSession_CreateWithoutSkillsSucceeds(t *testing.T) {
	apiKey := testutil.GetTestAPIKey(t)
	agentBay, err := agentbay.NewAgentBay(apiKey)
	if err != nil {
		t.Fatalf("Error initializing AgentBay client: %v", err)
	}

	params := agentbay.NewCreateSessionParams()
	result, err := agentBay.Create(params)
	if err != nil {
		t.Fatalf("Create failed: %v", err)
	}

	if !result.Success {
		t.Fatalf("Create returned success=false")
	}
	if result.Session == nil {
		t.Fatal("Expected non-nil session")
	}

	session := result.Session
	defer func() {
		_, _ = agentBay.Delete(session)
	}()
}
